src/pages/index.js

```diff
@@ -60,28 +107,122 @@ document.addEventListener("DOMContentLoaded", function () {
+  constants.avatarButton.classList.replace(
+    "profile__avatar-button",
+    "profile__avatar-button_disabled"
+  );
 });
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    (id) => handleDeleteClick(id),
+    (id) => handleLikeClick(id)
+  );
   return card.getView();
 }
 
-function handleImageClick(name, link) {
-  newPopupImage.open(name, link);
+function handleImageClick(cardData) {
+  newPopupImage.open(cardData);
+}
+
+function handleDeleteClick(card) {
+  popupAffirm.open();
+  popupAffirm.setSubmitAction(() => {
+    api
+      .deleteCards(card._id)
+      .then(() => {
+        popupAffirm.close();
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeClick(card) {
+  if (card.isLiked === true) {
+    api
+      .deleteCardLike(card._id)
+      .then((res) => {
+        card.isLiked = false;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .putCardLike(card._id)
+      .then((res) => {
+        card.isLiked = true;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
 }
 
 function handleProfileFormSubmit(formData) {
-  useInfo.setUserInfo(formData.title, formData.description);
-  editPopup.close();
+  constants.editProfileSubmitButton.textContent = "Saving...";
+  api
+    .updateProfileInfo(formData.title, formData.description)
+    .then((res) => {
+      useInfo.setUserInfo(res.name, res.about);
+      editPopup.close();
+      constants.editProfileSubmitButton.textContent = "Save";
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
 function handleCardFormSubmit(name, link) {
-  newCardList.addItem(name, link);
-  cardPopup.close();
-  constants.addCardForm.reset();
+  constants.addCardSubmitButton.textContent = "Saving...";
+  api
+    .createCards(name, link)
+    .then((res) => {
+      newCardList.addItem(res);
+      cardPopup.close();
+      constants.addCardForm.reset();
+      constants.addCardSubmitButton.textContent = "Save";
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleAvatarFormSubmit(inputObj) {
+  constants.avatarSubmitButton.textContent = "Saving...";
+  api
+    .updateAvatar(inputObj.link)
+    .then(() => {
+      useInfo.setAvatar(inputObj.link);
+      avatarPopup.close();
+      constants.avatarSubmitButton.textContent = "Save";
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors